langchain4j-core/src/main/java/dev/langchain4j/model/discovery/ModelDiscovery.java

```diff
@@ -0,0 +1,81 @@
+package dev.langchain4j.model.discovery;
+
+import dev.langchain4j.model.ModelProvider;
+import java.util.List;
+
+/**
+ * Represents a service that can discover available models from an LLM provider.
+ * This allows customers to list available models and their capabilities without
+ * needing to go to individual AI provider websites.
+ *
+ * <p>Similar to {@link dev.langchain4j.model.chat.ChatModel} and
+ * {@link dev.langchain4j.model.chat.StreamingChatModel}, each provider should
+ * implement this interface to enable model discovery.
+ *
+ * <p>Example usage:
+ * <pre>{@code
+ * OpenAiModelDiscovery discovery = OpenAiModelDiscovery.builder()
+ *     .apiKey(apiKey)
+ *     .build();
+ *
+ * List<ModelDescription> models = discovery.discoverModels();
+ *
+ * // Filter for specific models
+ * ModelDiscoveryFilter filter = ModelDiscoveryFilter.builder()
+ *     .types(Set.of(ModelType.CHAT))
+ *     .requiredCapabilities(Set.of(Capability.RESPONSE_FORMAT_JSON_SCHEMA))
+ *     .build();
+ * List<ModelDescription> chatModels = discovery.discoverModels(filter);
+ * }</pre>
+ *
+ * @see ModelDescription
+ * @see ModelDiscoveryFilter
+ */
+public interface ModelDiscovery {
```

Comment:
```suggestion
public interface ModelCatalog {
```
I am leaning more towards a "catalog", as it better reflects what this class is doing. WDYT?

Reply:
Also, "discovery" sounds more like a continuous process rather than an entity (noun), so it is not fitting well according to Java conventions

Reply:
I would refuse doing this by hand - but lets see, whether Claude can do this consistently. :-)

Reply:
Just in case, this refactoring is usually done in a few clicks with any modern IDE

Reply:
(without LLM)